### PR TITLE
Feature/update-manifest

### PR DESCRIPTION
**Pull request by changev**
## Background:
Manifest records a list of repositories for rackhd.
For example:

        {
            "branch": "master", 
            "commit-id": "38a0d873ffcc847745a0a4b586bc2675a1957ee7", 
            "repository": "https://github.com/RackHD/on-core.git"
        }
## Details
* Based on an exists manifest file or using the template empty manifest, this script receives commit sha and repo url to update the manifest file.

* After updates the new manifest file will be upload to bintray.

* After each updates the validation(every commit-id is not None) will be applied on manifest file. If validation succeeds, a property file will be created to trigger downstream BuildRelease job.

### CODE REVIEW

**Comment by yyscamper** [comment]
@changev : You forgot to rebase your branch.

**Comment by changev** [comment]
@panpan0000 @PengTian0  please review. Thanks.

**Comment by PengTian0** [comment]
:+1: After fix the comment.

File: manifest-build-tools/lib/RepositoryOperator.py [hunk 1]
```diff
@@ -108,7 +108,7 @@ def do_one_task(self, name, data, results):
         results['commands'] = []
         commands = results['commands']
         git = GitBit(verbose=False)
-        if 'credentials' in data:
+        if 'credentials' in data and data['credentials'] is not None:
```
Comment (panpan0000) [inline manifest-build-tools/lib/RepositoryOperator.py]: :-1: 
conflict with Tian's PR https://github.com/RackHD/on-tools/pull/59/files, which is more completed.
please align with it

File: manifest-build-tools/application/update_manifest.py [hunk 2]
```diff
@@ -0,0 +1,442 @@
...
+# Copyright 2015, EMC, Inc.
+
+"""
+Usage:
+./on-tools/manifest-build-tools/HWIMO-BUILD on-tools/manifest-build-tools/application/update_manifest.py \
+--repo "$url" \
+--branch "$branch" \
+--commit "$id" \
+--manifest_download_url https://dl.bintray.com/rackhd-mirror/binary/manifest/ \
+--git-credential https://github.com,GITHUB_CREDS \
```
Comment (panpan0000) [inline manifest-build-tools/application/update_manifest.py]: is git-credential still necessary ?
since the manifest file located in bintray ?

Comment (changev) [inline manifest-build-tools/application/update_manifest.py]: It'll clone the repo of the ```--repo "$url"  --branch "$branch" -commit "$id" ```
to check if the parameters are valid.
So, though rackhd is a public repo, the ```--git-credential``` is still needed.

File: manifest-build-tools/lib/RepositoryOperator.py (hunk 1 above)
Comment (changev) [inline manifest-build-tools/lib/RepositoryOperator.py]: I'll remove this change

**Comment by changev** [comment]
@panpan0000 @PengTian0 
PR has been updated.

File: manifest-build-tools/application/update_manifest.py (hunk 2 above)
Comment (panpan0000) [inline manifest-build-tools/application/update_manifest.py]: if the existing functions ```repo_operator.setup_gitbit```  and ```Manifest(path_name, self.__git_credentials)``` needs git-credential during initialization ,  then :+1: now.
but I still prefer it to be optional , + @PengTian0   to be aware

**Comment by panpan0000** [comment]
make :+1: more visible

File: manifest-build-tools/application/update_manifest.py (hunk 2 above)
Comment (PengTian0) [inline manifest-build-tools/application/update_manifest.py]: The parameter git_credential is optional for the class ```RepoOperator``` and ```Manifest```.
I think the parameter should be optional to the script update_manifest.

**Pull request merged**